### ISSUE TITLE
[Bug-Candidate]: False positive uses timestamp for array length comparison

### ISSUE THREAD

**Issue by pemmenegger** [crytic/slither#1436]
### Describe the issue:

A Check: ```timestamp``` gets triggered when comparing to struct array length.

### Code example to reproduce the issue:

Inside my contract:

    struct Post {
        uint256 id;
        uint256 timeCreated; // block.timestamp will be assigned here on creation
        address contributor;
        uint256 distributedSocialTokens;
    }
    
    Post[] public posts;
    
    function _createPost() internal {
          posts.push(Post(posts.length, block.timestamp, msg.sender, 0));
     }
    
    function _likePost(uint256 postIdx) internal onlyMember {
        require(postIdx < posts.length, "Post does not exist");
        /* ... */
    }

### Version:

0.8.3

### Relevant log output:

```shell
AFeed._likePost(uint256) (contracts/feeds/AFeed.sol#58-93) uses timestamp for comparisons
        Dangerous comparisons:
        - require(bool,string)(postIdx < posts.length,Post does not exist)(contracts/feeds/AFeed.sol#59)
```

**Comment by 0xalpharush** [comment]
I believe it's related to this [limitation](https://github.com/crytic/slither/issues/365) in how structures are represented. I will see what it takes to resolve this cc @montyly

**Comment by montyly** [comment]
Yes unfortunately this is due to our current IR, and to solve it correctly we need to improve it.
